scipy/_lib/_util.py

```diff
@@ -397,3 +397,63 @@ def __call__(self, func, iterable):
             # wrong number of arguments
             raise TypeError("The map-like callable must be of the"
                             " form f(func, iterable)")
+
+
+# _normalize_axis_index() is a reimplementation of the private numpy function
+# numpy.core.multiarray.normalize_axis_index.  If that function is made public,
+# it can be used instead of this one.
+
+def _normalize_axis_index(axis, ndim, msg_prefix=None):
+    """
+    Normalizes an axis index, `axis`, such that it is a valid positive index
+    into the shape of array with `ndim` dimensions. Raises an `AxisError` with
+    an appropriate message if this is not possible.
+
+    Parameters
+    ----------
+    axis : int
+        The un-normalized index of the axis. Can be negative
+    ndim : int
+        The number of dimensions of the array that `axis` should be normalized
+        against
+    msg_prefix : str
+        A prefix to put before the message, typically the name of the argument
+
+    Returns
+    -------
+    normalized_axis : int
+        The normalized axis index, such that `0 <= normalized_axis < ndim`
+
+    Raises
+    ------
+    AxisError
+        If the axis index is invalid, when `-ndim <= axis < ndim` is false.
+
+    Examples
+    --------
+    >>> normalize_axis_index(0, ndim=3)
+    0
+    >>> normalize_axis_index(1, ndim=3)
+    1
+    >>> normalize_axis_index(-1, ndim=3)
+    2
+
+    >>> normalize_axis_index(3, ndim=3)
+    Traceback (most recent call last):
+    ...
+    AxisError: axis 3 is out of bounds for array of dimension 3
+    >>> normalize_axis_index(-4, ndim=3, msg_prefix='axes_arg')
+    Traceback (most recent call last):
+    ...
+    AxisError: axes_arg: axis -4 is out of bounds for array of dimension 3
```

Comment:
```suggestion
    Traceback (most recent call last):
      ...
    AxisError: axes_arg: axis -4 is out of bounds for array of dimension 3
```
Looks more like the real output